livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py

```diff
@@ -626,6 +628,16 @@ def create_azure_client(
             tool_choice=tool_choice,
         )
 
+    @wraps(AsyncCompletions.create)
+    def create_chat_completions_stream(self, **kwargs):
```

Comment:
Is the goal to be able to edit the parameters of the LLM inside the VoicePipelineAgent?

Reply:
in general, yes
we'd like to inject custom values to the corresponding openai API in the descendant class, which are either not implemented in the agent for now or that contain customized logic, that would probably not make much sense for this particular (generic) agent implementation

Reply:
Could the before_llm_cb callback inside PipelineAgent do the work? You can return an LLMStream from there (so you can use chat with any parameter you want)

Reply:
I assume with the recent changes in main this PR becomes obsolete.
The original intention was to customize parameters passed to the OpenAI completion creation API rather than these that the `chat` one exposes. 

For now I think it would make sense to close this PR since we anyway override `_main_task` (now called `_run`) of the `llm.LLMStream` class

Reply:
Interesting, I think this [PR](https://github.com/livekit/agents/pull/1210) may be interesting for it

Reply:
most likely it won't because  `_extra_options` cannot override existing arguments